src/sentry/static/sentry/app/components/issueDiff/groupingDiff.tsx

```diff
@@ -0,0 +1,57 @@
+import {t} from 'app/locale';
+import {EventGroupComponent, EventGroupInfo, EventGroupVariant} from 'app/types';
+
+function renderGroupingInfo(groupingInfo: EventGroupInfo): string[] {
+  return Object.values(groupingInfo)
+    .map(renderGroupVariant)
+    .flat();
+}
+
+function renderGroupVariant(variant: EventGroupVariant): string[] {
+  const title = [t('Type: %s', variant.type)];
+
+  if (variant.hash) {
+    title.push(t('Hash: %s', variant.hash));
+  }
+
+  if (variant.description) {
+    title.push(t('Description: %s', variant.description));
+  }
+
+  const rv = [title.join('\n')];
+
+  if (variant.component) {
+    rv.push(renderComponent(variant.component).join('\n'));
+  }
+
+  return rv;
+}
+
+function renderComponent(component: EventGroupComponent): string[] {
+  if (!component.contributes) {
+    return [];
+  }
+
+  let title = component.name || '';
+  if (component.hint) {
+    title += ` (${component.hint})`;
+  }
```

Comment:
```suggestion
 const {name = ''} = component;
 const title = component.hint ? `${name} (${component.hint})` : name;
```
but is it fine to have extra space if there's no name?

Reply:
good catch, fixed